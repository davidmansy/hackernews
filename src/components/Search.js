import React from 'react';

const Search = ({ query, onSearchChange, children }) => {
  return (
    <form>
      {children}
      <input
        type="text"
        value={query}
        onChange={e => onSearchChange(e.target.value)}
      />
    </form>
  );
};

export default Search;
