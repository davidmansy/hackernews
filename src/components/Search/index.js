import React, { Component } from 'react';

class Search extends Component {
  componentDidMount() {
    this.input.focus();
  }

  render() {
    const { query, onChange, children, onSubmit } = this.props;
    return (
      <form onSubmit={onSubmit}>
        <input
          type="text"
          value={query}
          onChange={e => onChange(e.target.value)}
          ref={node => (this.input = node)}
        />
        <button type="submit">{children}</button>
      </form>
    );
  }
}

export default Search;
