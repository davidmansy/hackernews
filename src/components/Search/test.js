import React from 'react';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';
import Search from './';

function createNodeMock(element) {
  if (element.type === 'input') {
    return {
      focus() {}
    };
  }
  return null;
}

describe('Search', () => {
  it('renders without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(<Search>Search</Search>, div);
    ReactDOM.unmountComponentAtNode(div);
  });

  it('snapshot', () => {
    const options = { createNodeMock };
    const component = renderer.create(<Search>Search</Search>, options);
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
