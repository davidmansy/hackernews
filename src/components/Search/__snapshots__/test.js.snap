// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Search snapshot 1`] = `
<form
  onSubmit={undefined}
>
  <input
    onChange={[Function]}
    type="text"
    value={undefined}
  />
  <button
    type="submit"
  >
    Search
  </button>
</form>
`;
